compiler/rustc_hir_typeck/src/lib.rs

```diff
@@ -265,7 +265,10 @@ fn typeck_with_fallback<'tcx>(
 
     for (ty, span, code) in fcx.deferred_sized_obligations.borrow_mut().drain(..) {
         let ty = fcx.normalize(span, ty);
-        fcx.require_type_is_sized(ty, span, code);
+        // ScalableSIMD: Justify this.
```

Comment:
Needs a TODO or a very detailed explanation here

Reply:
This check should be moved to the call sites of `require_type_is_sized_deferred`.